lib/ecto_ldap/adapter.ex

```diff
@@ -230,11 +177,81 @@ defmodule Ecto.Ldap.Adapter do
   def translate_value(%Ecto.Query.Tagged{value: value}), do: value
   def translate_value(other), do: other
 
-  def load(:id, value), do: {:ok, value}
+  def execute(_repo, query_metadata, prepared, params, preprocess, _options) do
+
+    search_response   = search(prepared, params)
+    fields            = ordered_fields(query_metadata.sources)
+    count             = count_fields(query_metadata.select.fields, query_metadata.sources)
+
+    {:ok, {:eldap_search_result, results, []}} = search_response
+
+    result_set = results
+    |> Enum.map(&create_rows(&1))
+    |> Enum.map(&prune_columns(&1, fields))
+    |> Enum.map(&generate_models(&1, preprocess, count))
+
+    {count, result_set}
+  end
+
+  def ordered_fields(sources) do
+    {_, model} = elem(sources, 0)
+    model.__schema__(:fields)
+  end
+
+  def count_fields(fields, sources) do
+    Enum.map fields, fn
+      {:&, _, [idx]} = field ->
+        {_source, model} = elem(sources, idx)
+        {field, length(model.__schema__(:fields))}
+      field ->
+        {field, 0}
+    end
+  end
+
+  def create_rows({:eldap_entry, dn, attributes}) when is_list(attributes) do
```

Comment:
Is there a specific reason this was changed from `process_entry` back to `create_rows` ? Because we don't have any rows and we're not creating any...